resources/developer-mode.mjs

```diff
@@ -63,33 +68,56 @@ export function createUIForWarmupSuite() {
     return label;
 }
 
-export function createUIForIterationCount() {
-    let range = document.createElement("input");
-    range.type = "range";
-    range.id = "iteration-count";
-    range.min = 1;
-    range.max = 20;
-    range.value = params.iterationCount;
-
-    let label = document.createElement("label");
-    let countLabel = document.createElement("span");
-    countLabel.textContent = params.iterationCount;
-    label.append("iterations: ", countLabel, document.createElement("br"), range);
-
-    range.oninput = () => {
-        countLabel.textContent = range.value;
+function createUIForIterationCount() {
+    const { range, label } = createTimeRangeUI("Iterations: ", params.iterationCount, "#", 1, 100);
+    range.onchange = () => {
+        params.iterationCount = parseInt(range.value);
+        updateURL();
     };
+    return label;
+}
 
+function createUIForWarmupBeforeSync() {
+    const { range, label } = createTimeRangeUI("Warmup time: ", params.warmupBeforeSync);
     range.onchange = () => {
-        params.iterationCount = parseInt(range.value);
+        params.warmupBeforeSync = parseInt(range.value);
         updateURL();
     };
+    return label;
+}
 
+function createUIForSyncStepDelay() {
+    const { range, label } = createTimeRangeUI("Sync step delay: ", params.waitBeforeSync);
+    range.onchange = () => {
+        params.waitBeforeSync = parseInt(range.value);
+        updateURL();
+    };
     return label;
 }
 
+function createTimeRangeUI(labelText, initialValue, unit = "ms", min = 0, max = 1000) {
+    const range = document.createElement("input");
+    range.type = "range";
+    range.min = min;
+    range.max = max;
+    range.value = initialValue;
+
+    const label = document.createElement("label");
+    const rangeLabel = document.createElement("span");
```

Comment:
nit: the name "label" can be confusing in this context, I believe it should be renamed, for example to `rangeValueText`?

Reply:
I wonder if `rangeLabel` should be outside of the `label` element (it's not labeling the input), but this would need more CSS changes that we're willing to do here, I believe.

Reply:
Renamed the variable to rangeValueAndUnit, let's maybe move the rest for later.